tests/rptest/tests/isolated_decommissioned_node_test.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from rptest.services.admin import Admin
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+from rptest.tests.prealloc_nodes import PreallocNodesTest
+from rptest.util import firewall_blocked
+from confluent_kafka import (Producer, KafkaException)
+
+import time
+
+
+def on_delivery(err, msg):
+    if err is not None:
+        raise KafkaException(err)
+
+
+class IsolatedDecommissionedNodeTest(PreallocNodesTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3), )
+
+    def __init__(self, test_context):
+        super(IsolatedDecommissionedNodeTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             node_prealloc_count=3)
+
+        self.internal_port = 33145
+        self.admin = Admin(self.redpanda)
+
+    @cluster(num_nodes=3)
+    def simple_produce_test(self):
```

Comment:
Can we test the opposite case, where we disable the isolation logic (by setting some very high timeout), and then demonstrate a client hanging?  That would give more confidence that the isolation handling code is definitely making a difference.

Otherwise the test might be passing just because the client is helpful and trying a different broker for us, or because the topic's leadership was on a different node + still up to date in the metadata on the isolated node.

Reply:
Yes, I am working on more test, which will not work without new logic! Just added smth simple to see that logic works